src/main/java/network/brightspots/rcv/EditableTableCell.java

```diff
@@ -0,0 +1,216 @@
+import javafx.scene.input.KeyCode;
+import javafx.util.Callback;
+import javafx.util.StringConverter;
+import javafx.util.converter.DefaultStringConverter;
+
+/**
+ * A class containing a {@link TableCell} implementation that draws a
+ * {@link TextField} node inside the cell. If the TextField is
+ * left, the value is committed.
+ */
+
+public class EditableTableCell<S, T> extends TableCell<S, T> {
+  private TextField textField;
+  private boolean escapePressed = false;
+  private TablePosition<S, ?> tablePos = null;
+
+  private final ObjectProperty<StringConverter<T>> converter =
+      new SimpleObjectProperty<>(this, "converter");
+
+  /**
+   * Provides a TextFieldTableCell that allows editing of the cell content with any type.
+   * This method will only work on any type, but you'll need to provide a StringConverter.
+   */
+  public EditableTableCell(StringConverter<T> converter) {
+    this.getStyleClass().add("text-field-table-cell");
+    setConverter(converter);
+  }
+
+  /**
+   * Provides a TextField that allows editing of the cell content when
+   * the cell is double-clicked, or when TableView.edit is called.
+   * This method will only work on TableColumn instances which are of
+   * type String.
+   */
+  public static <S> Callback<TableColumn<S, String>, TableCell<S, String>> forTableColumn() {
+    return forTableColumn(new DefaultStringConverter());
+  }
+
+  /**
+   * Provides a TextField that allows editing of the cell content when
+   * the cell is double-clicked, or when TableView.edit is called.
+   * This method will only work on any type, but you'll need to provide a StringConverter.
+   */
+  public static <S, T> Callback<TableColumn<S, T>, TableCell<S, T>> forTableColumn(
+      final StringConverter<T> converter) {
+    return list -> new EditableTableCell<>(converter);
+  }
+
+  private ObjectProperty<StringConverter<T>> converterProperty() {
+    return converter;
+  }
+
+  private StringConverter<T> getConverter() {
+    return converterProperty().get();
+  }
+
+  private void setConverter(StringConverter<T> value) {
+    converterProperty().set(value);
+  }
+
+  @Override
+  public void startEdit() {
+    if (isEditable()
+        && getTableView().isEditable()
+        && getTableColumn().isEditable()) {
+      super.startEdit();
+
+      if (isEditing()) {
+        // Update members
+        escapePressed = false;
+        tablePos = getTableView().getEditingCell();
+
+        // Update the textfield
+        if (textField == null) {
+          textField = getTextField();
+        }
+        textField.setText(getItemText());
+
+        // Update the cell graphics
+        setText(null);
+        setGraphic(textField);
+        textField.selectAll();
+
+        // requesting focus so that key input can immediately go into the
+        // TextField (see RT-28132)
+        textField.requestFocus();
+      }
+    }
+  }
+
+  @Override
+  public void commitEdit(T newValue) {
```

Comment:
Hm, that feels pretty strange too -- if you:
1. Begin edit
2. Hit tabulate

Should it cancel or commit your edit? I can see arguments both ways. I've opted to just disable the menu bar and tab bar until the edit is committed or canceled.

Reply:
Yeah, that's a good point, and I like the direction you went. Could you modify this further to disable everything outside the table? For example, if I edit a cell, I can still click the auto-load button, which re-enables the menu at the top.

Reply:
Done -- there's not an existing clean way to grab everything outside of the table since many of the buttons aren't hooked up to variables in the class, so the best way is to traverse through the scene graph and find things to disable.

Let me know if you're okay with how I did it!

Reply:
On the CVR Files tab, it doesn't disable the buttons there. There's another edge case where you can edit a cell and keep it open, then go through and add another CVR. As soon as you click add or clear, it commits the edit down below (so it's making the decision you mentioned above that's kind of ambiguous).

Can we just disable an entire box and everything in it (in this case the "Add CVR File" box), including all buttons and inputs? If not, I guess we just need more bespoke code targeting the buttons on this pane as well.

Reply:
I cannot disable the entire panel since that would disable the edit button too. I think I have an idea for a cleaner solution -- I might be able to tag CSS selectors for everything that needs to be disabled, and then just loop over that selector. Will look into this.

Reply:
Done -- I now use CSS style selectors to identify what to disable.

Reply:
This is a great solution, and exactly what I was hoping for!

The only other corner case I can find now is, if you're editing and you click the X to close the program, it won't commit the edit and it'll end up effectively ignoring the edit if you save it. Probably fine to leave it this way though.